src/classes/BGE_ConfigurationWizard_CTRL.cls

```diff
@@ -43,87 +43,70 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
-                    model.id = (String)recordDetails.get('Id');
-                    model.description = (String)recordDetails.get(descriptionField);
-                    model.requireTotalMatch = (Boolean)recordDetails.get('RequireTotalMatch__c');
-                    model.activeFields = (String)recordDetails.get('Active_Fields__c');
-                    model.batchProcessSize = (Decimal)recordDetails.get('Batch_Process_Size__c');
-                    model.donationDateRange = (Decimal)recordDetails.get('Donation_Date_Range__c');
-                    model.donationMatchingBehavior = (String)recordDetails.get('Donation_Matching_Behavior__c');
-                    model.donationMatchingClass = (String)recordDetails.get('Donation_Matching_Implementing_Class__c');
-                    model.donationMatchingRule = ((String)recordDetails.get('Donation_Matching_Rule__c')).split(';');
-                    model.postProcessClass = (String)recordDetails.get('Post_Process_Implementing_Class__c');
-                    model.processUsingScheduledJob = (Boolean)recordDetails.get('Process_Using_Scheduled_Job__c');
-                    model.runOpportunityRollupsWhileProcessing = (Boolean)recordDetails.get('Run_Opportunity_Rollups_while_Processing__c');
-
-                    if (sObjectName == 'DataImportBatch__c') {
-                        model.expectedCount = Integer.valueOf(recordDetails.get('Expected_Count_of_Gifts__c'));
-                        model.expectedTotal = (Decimal)recordDetails.get('Expected_Total_Batch_Amount__c');
-                    }
-                } catch (Exception ex) {
-                    throw new AuraHandledException('ERROR: ' + ex.getMessage());
-                }
-            } else {
-                throw new AuraHandledException('ERROR: Only Batch_Template__c and DataImportBatch__c are supported objects.');
+            try {
+                List<String> fields = new List<String>{
+                    'Batch_Description__c',
+                    'Name',
+                    'Id',
+                    'RequireTotalMatch__c',
+                    'Active_Fields__c',
+                    'Batch_Process_Size__c',
+                    'Run_Opportunity_Rollups_while_Processing__c',
+                    'Contact_Matching_Rule__c',
+                    'Donation_Matching_Behavior__c',
+                    'Donation_Matching_Implementing_Class__c',
+                    'Donation_Matching_Rule__c',
+                    'Donation_Date_Range__c',
+                    'Post_Process_Implementing_Class__c',
+                    'Process_Using_Scheduled_Job__c',
+                    'Expected_Count_of_Gifts__c',
+                    'Expected_Total_Batch_Amount__c'
+                };
+
+                String soql = 'SELECT ' +
+                    String.join(fields, ', ') +
+                    ' FROM DataImportBatch__c'+
+                    ' WHERE Id=:recordId';
+
+                SObject recordDetails = Database.query(soql);
```

Comment:
We can cast recordDetails to a DataImportBatch__c object now, and don't have to do the .get and type casting below. Yay for easier-to-read code!